lib/grape/idempotency.rb

```diff
@@ -113,24 +118,35 @@ def get_from_cache(idempotency_key)
         JSON.parse(value)
       end
 
-      def store_request(idempotency_key, path, params, status, request_id, response)
+      def store_processing_request(idempotency_key, path, params, request_id)
         body = {
           path: path,
           params: params,
-          status: status,
           original_request: request_id,
-          response: response
-        }.to_json
+          processing: true
+        }
 
-        result = storage.set(key(idempotency_key), body, ex: configuration.expires_in, nx: true)
+        result = storage.set(key(idempotency_key), body.to_json, ex: configuration.expires_in, nx: true)
 
         if !result
-          return store_request(random_idempotency_key, path, params, status, request_id, response)
-        else
-          return idempotency_key
+          # This should not be reach as we are checking if there is a processing request before
```

Comment:
I think is not possible to reach here a point where a processing document is found here since we check that previously but was weird not checking the result

Reply:
It can happen if we have two parallels requests making the first call with the same idempotency key at the same exact time. The first will store the processing request but as we have the `nx:true`, the second one will not be stored. What about checking the params and path of the already stored processing request and if params does not match return `409 - Conflict` and if the params and path match return `102 - Processing`?

Reply:
so you mean something like this?:

```suggestion
          if!result
           entry = JSON.parse(storage.get(key(idempotency_key))
           if conflict?(entry, path, params)
             conflict_response
           else
             processing_response
           end
          end
```

Reply:
Yes, but after reading the https://datatracker.ietf.org/doc/html/draft-ietf-httpapi-idempotency-key-header-03 maybe we should return always conflict here?